### PR TITLE
1293 async pool map call pipeline v3

### PR DESCRIPTION
**Pull request by HustonMmmavr**
What has been done? Why? What problem is being solved?

I didn't forget about

- [ ] Tests
- [ ] Changelog
- [ ] Documentation

Close #???

### CODE REVIEW

**Comment by rosik** [comment]
Not a related question, but I'm interested.

```lua

local function _pack_values(maps, uri, ...)
    -- Spread call results across tables
    for i = 1, select('#', ...) do
        local v = select(i, ...)
        maps[i] = maps[i] or {}
        maps[i][uri] = v
    end
end
```

Can it be optimized with some kind of loop unrolling? e.g.

```lua
local function _pack_values(maps, uri, ok, err, ...)
```

File: cartridge/pool.lua [hunk 1]
```diff
@@ -203,29 +195,57 @@ local function map_call(fn_name, args, opts)
...
+    local deadline = fiber.clock() + timeout
+
+    for _, uri in ipairs(opts.uri_list) do
+        local conn, err = connect(uri, {wait_connected = false})
+        if conn == nil then
+            _pack_values(maps, uri, nil, err)
+        elseif conn:is_connected() then
+            local future, err = errors.netbox_call(
+                conn, fn_name, args, {is_async = true}
+            )
```
Comment (rosik) [inline cartridge/pool.lua:212]: Can it ever raise an error?

File: cartridge/pool.lua [hunk 2]
```diff
@@ -203,29 +195,57 @@ local function map_call(fn_name, args, opts)
...
 
-    local retmap, errmap = unpack(maps)
+    for uri, future in pairs(futures) do
+        local timeout = deadline - fiber.clock()
+        if timeout < 0 then
+            timeout = 0
+        end
+
+        _pack_values(maps, uri,
+            errors.netbox_wait_async(future, timeout, uri, fn_name)
```
Comment (rosik) [inline cartridge/pool.lua:243]: Excess args 3, 4

File: cartridge/pool.lua (hunk 1 above)
Comment (HustonMmmavr) [reply cartridge/pool.lua:212]: For ex. connection became broken (but it's unlikely).
Also, I think it's better to wrap all `netbox` methods.

**Comment by rosik** [comment]
Merged in #1412 instead.

**Pull request closed without merging**